testing/integration/tests/fil-ipld-actor/src/actor.rs

```diff
@@ -13,20 +21,44 @@ pub fn invoke(_: u32) -> u32 {
         )
     }));
 
+    test_open_block();
     test_read_block();
+    test_create_block();
+    test_stat_block();
+    test_link_block();
 
     #[cfg(coverage)]
     sdk::debug::store_artifact("ipld_actor.profraw", minicov::capture_coverage());
     0
 }
 
+fn test_open_block() {
+    let test_bytes = gen_test_bytes(1 << 10);
+
+    unsafe {
+        let cid = sdk::ipld::put(0xb220, 32, DAG_CBOR, &test_bytes).unwrap();
+
+        // The cid should be valid
+        let mut buf = [0u8; MAX_CID_LEN];
+        cid.write_bytes(&mut buf[..])
+            .expect("CID encoding should not fail");
+        sdk::sys::ipld::block_open(buf.as_mut_ptr()).expect("shouldwork");
+
+        // Test for invalid Cid
+        buf.fill(0);
+        let res = sdk::sys::ipld::block_open(buf.as_mut_ptr());
+        assert_eq!(res, Err(ErrorNumber::IllegalArgument));
+
+        // Test for invalid Cid pointer
+        let res = sdk::sys::ipld::block_open("invalid ptr".as_ptr());
+        assert_eq!(res, Err(ErrorNumber::IllegalArgument));
+
+        // TODO (fridrik): Test for very large cid
```

Comment:
Yeah, this'll be a bit annoying as you'll have to manually construct it. Also not something that's super critical to test.